packages/client/src/app/pages/forms/edit/builder/tabs/layout/field-list/implementations/base-fields/base-fields.tsx

```diff
@@ -1,37 +1,89 @@
 
-import { FieldGroup } from '../../field-group/field-group';
+import { ManagerFieldGroup } from '../../field-group/field-group';
 import { LoaderFieldGroup } from '../../field-group/field-group.loader';
-import { useSelectSearchedFields } from '../../hooks/use-select-searched-fields';
+import { List } from '../../field-group/field-group.styles';
+import { useSelectSearchedGroups } from '../../hooks/use-select-searched-fields';
 
+import { useCreateModal } from './modal/use-create-modal';
+import { GroupName, GroupWrapper } from './base-fields.styles';
 import { FieldItem } from './field-item';
 
 const title = translate('Field Types');
 
 export const BaseFields: React.FC = () => {
-  const select = useSelectSearchedFields();
-  const { data, isFetching, isError, error } = useFetchFieldTypes({ select });
+  const select = useSelectSearchedGroups();
+  const {
+    data: groupsData,
+    isFetching: groupsFetching,
+    isError: groupsIsError,
+    error: groupsError,
+  } = useFetchGroups({ select });
+  const {
+    data: fieldTypesData,
+    isFetching: fieldTypesFetching,
+    isError: fieldTypesIsError,
+    error: fieldTypesError,
+  } = useFetchFieldTypes();
+  const openModal = useCreateModal();
 
-  if (!data && isFetching) {
-    return <LoaderFieldGroup words={[50, 70]} items={16} />;
-  }
+  const loading = groupsFetching || fieldTypesFetching;
+  const hasError = groupsIsError || fieldTypesIsError;
 
-  if (isError) {
-    return <ErrorBlock>{error.message}</ErrorBlock>;
+  if (loading && (!groupsData || !fieldTypesData)) {
+    return <LoaderFieldGroup words={[50, 70]} items={16} />;
   }
 
-  if (!data.length) {
-    return null;
+  if (hasError) {
+    return (
+      <ErrorBlock>{groupsError.message || fieldTypesError.message}</ErrorBlock>
+    );
   }
 
   return (
-    <FieldGroup title={title}>
-      {data.map((fieldType) => (
-        <FieldItem key={fieldType.typeClass} fieldType={fieldType} />
+    <ManagerFieldGroup
+      button={{
+        icon: <EditIcon />,
+        title: translate('Edit Manager'),
+        onClick: openModal,
+      }}
+      editionIsPro={config.editions.is(Edition.Pro)}
+      title={title}
+    >
+      {groupsData.groups.grouped?.map((group) => (
+        <GroupWrapper key={group.uid} color={group.color}>
+          {group.types.length > 0 && (
+            <>
+              {group.label && <GroupName>{group.label}</GroupName>}
+              <List>
+                {group.types.map((type) => {
+                  const fieldType = fieldTypesData.find(
+                    (item) => item.typeClass === type
+                  );
+                  return <FieldItem key={type} fieldType={fieldType} />;
+                })}
+              </List>
+            </>
+          )}
+        </GroupWrapper>
       ))}
-    </FieldGroup>
+
+      {groupsData?.types && (
+        <List>
+          {groupsData?.types.map((type) => {
+            const fieldType = fieldTypesData.find(
```

Comment:
I noticed the hook; it uses 'useFetchFieldTypes' every time and looks for the 'typeClass', Since there are many fields, I decided to make the request just once<useFetchFieldTypes>. The '<LoaderFieldGroup words={[50, 70]} items={16} />' will work nicely without needing to set it for every single item."